### ISSUE TITLE
gcc flags don't work when using ceedling test:*

### ISSUE THREAD

**Issue by lafonso-gnrc** [ThrowTheSwitch/Ceedling#882]
Hello,

To give context, I have a set of unit tests where one function was missing the return statement.

Somehow this passes in unit testing but it fails when running gcov (as in, it runs gcov fine but reports failures) which is weird. I'd expect both to behave the same but that's the main issue I have now.

Regardless, I decided to turn on the plugin ``` warnings_report ``` to be easier to check and add the flags

```
:flags:
  :release:
    :compile:
      :*:
        - -std=c99
        - -Wall
        - -Werror=return-type
        #- -Werror 
  :test:
    :compile:
      :*:
        - -std=c99
        - -Wall
        - -Werror=return-type
        #- -Werror
  :gcov:
      :compile:  
        :*:
          - -std=c99
          - -Wall
          - -Werror=return-type
```

So now I at least have gcov with the same flags, although I still get the different behaviours.
And it throws errors when it's missing return statements, good enough.

But, it doesn't work when doing ``` ceedling clobber test:* ``` for whatever test I want.
It seems the flags are not used in that situation.

I've tried ``` ceedling test:path[dir] ``` and that one takes the flags.

Any advice of what I could be doing wrong to not get compilation flags when running individual tests?

**Comment by mvandervoord** [comment]
Hi. Are you running 0.31 or one of the 0.32 release candidates?  :)

**Comment by lafonso-gnrc** [comment]
hello, 

I am using a snapshot, ceedling-0.32.0-164e950.

**Comment by mvandervoord** [comment]
If you run as `ceedling --verbosity=debug clobber test:blah` where `blah` is a filename you want to run, it's not using the specified flags when compiling `blah.c`?

**Comment by lafonso-gnrc** [comment]
```--verbosity=debug``` doesn't seem to work so I used ```verbosity[4] ```

It does not show the specified flags. Double checking with ```test:path[]``, I can seel all the flags. I can see this at the start

```gcc.exe -std=c99 -Wall -Werror=return-type ```

**Comment by mvandervoord** [comment]
Thanks... and sorry about the wrong verbosity flag. It just changed recently and I wasn't sure which you had. 

This definitely looks to be a bug. I don't know if you have the bandwidth to help debug it, but if you do, can you try this:

In the Ceedling gem (depending on your OS, you might be able to find it using `which ceedling`), you want to open the file `rules_tests.rake`. Somewhere around line 30, you'll see this:

```

namespace TEST_SYM do
  TOOL_COLLECTION_TEST_RULES = {
    #:context        => TEST_SYM,    <---- yours doesn't start with # but try adding it and saving this file
    :test_compiler  => TOOLS_TEST_COMPILER,
    :test_assembler => TOOLS_TEST_ASSEMBLER,
    :test_linker    => TOOLS_TEST_LINKER,
    :test_fixture   => TOOLS_TEST_FIXTURE
  }
```

See if that makes any difference to your setup?

**Comment by lafonso-gnrc** [comment]
I'm running it on windows because IAR and avoid switching between filesystems. ```gem which ceedling``` works for it.

This is what mine looks like initially
```
namespace TEST_SYM do
  TOOL_COLLECTION_TEST_RULES = {
    :test_compiler  => TOOLS_TEST_COMPILER,
    :test_assembler => TOOLS_TEST_ASSEMBLER,
    :test_linker    => TOOLS_TEST_LINKER,
    :test_fixture   => TOOLS_TEST_FIXTURE
  }
  ```

Changed to this and ran ceedling again. can-t really tell any difference in behavior or flags in verbosity.
```
namespace TEST_SYM do
  TOOL_COLLECTION_TEST_RULES = {
    :context        => TEST_SYM,
    :test_compiler  => TOOLS_TEST_COMPILER,
    :test_assembler => TOOLS_TEST_ASSEMBLER,
    :test_linker    => TOOLS_TEST_LINKER,
    :test_fixture   => TOOLS_TEST_FIXTURE
  }
  ```

**Comment by mvandervoord** [comment]
Thanks. You can switch it back. I'm going to try to replicate your issue here and see if I can figure out what's going on. :)

**Comment by lafonso-gnrc** [comment]
thank you, I appreciate it

**Comment by mvandervoord** [comment]
Are you in a position to update to the latest release candidate? I don't see anywhere that we should have fixed this issue specifically, but I also am unable to reproduce it using the latest release candidate.

**Comment by lafonso-gnrc** [comment]
I might get a chance to try this at some point, maybe only over the weekend

**Comment by lafonso-gnrc** [comment]
I tried doing this on WSL to not break my windows build environment. It had already a ceedling version installed.

I've done ```sudo gem install --local <path> ``` with release candidate [0.32.0-25f0aa4]. It doesn't seem able to identify the project as a ceedling project and I get these errors when doing ```ceedling help```
```
🪲 ERROR: Plugin 'stdout_pretty_tests_report' not found in built-in or project Ruby load paths. Check load paths and plugin naming and path conventions.
🪲 ERROR: Plugin 'warnings_report' not found in built-in or project Ruby load paths. Check load paths and plugin naming and path conventions.
```

I got to try this in a fresh install and project at some point. In general it seems like I randomly get features disabled/broken depending on the release candidate I try, probably due to already have previously installed versions.

**Comment by mkarlesky** [comment]
Hi, @lafonso-gnrc. Did you make any progress or resolve this? Quite a bit has changed including a great many bug fixes since you first reported this issue. I'd like to either close the issue or fix what sounds like a potentially serious bug before 1.0.0 ( previously 0.32) is officially released.

**Comment by lafonso-gnrc** [comment]
hello, I will have to try this out. I haven't updated from that one snapshot yet, last time I checked, parametrization from Unity (TEST_CASE AND TEST_RANGE) was not working yet so I haven't really checked much.

I'll see if I can try this on the weekend

**Comment by LuisAfonso95** [comment]
Hello, finally came around to try this. Sorry for the delay.

I've installed a fresh WSL Ubuntu. Of note, I had to install latest release of ceedling before installing the snapshot.
I am using snapshot 1.0.0-13d0e3d.

Created a new ceedling project and new module (named thingie). updated the project settings with
```
:flags:
   :test:
     :compile:
      :*:
         - -Werror=return-type
```

I've added a function that returns an int with the missing return statement.
Running both ```ceedling test:all``` and ```ceedling test:test_thingie``` both give me the error ``` error: control reaches end of non-void function [-Werror=return-type]``` .

So I believe this is fixed :)
